test/com/amazon/ion/impl/bin/WriteBufferTest.java

```diff
@@ -959,4 +959,118 @@ public void shiftBytesLeftAcrossBufferBlocksExclusively() throws IOException {
         buf.shiftBytesLeft(5, 5);
         assertBuffer("012345BCDEF".getBytes());
     }
+
+    @Test
+    public void shiftBytesLeftAcrossBufferBlocksEmptyingLastBlock() throws IOException {
+        assertEquals(11, ALLOCATOR.getBlockSize());
+        // The "B" is the first and only byte in the second block.
+        // Shifting 1 byte left by one empties the last block.
+        buf.writeBytes("0123456789AB".getBytes());
+        buf.shiftBytesLeft(1, 1);
+        assertBuffer("0123456789B".getBytes());
+    }
+
+    @Test
+    public void shiftEntireBlock() throws IOException {
+        assertEquals(11, ALLOCATOR.getBlockSize());
+        // The buffer contains two full blocks
+        buf.writeBytes("0123456789|ABCDEFGHIJ|".getBytes());
+        // We shift an entire block left by the block size
+        buf.shiftBytesLeft(11, 11);
+        assertBuffer("ABCDEFGHIJ|".getBytes());
+    }
+
+    @Test
+    public void shiftBytesLeftByMoreThanTheBlockSize() {
+        assertEquals(11, ALLOCATOR.getBlockSize());
+        // We have 5 blocks' worth of data
+        buf.writeBytes("0123456789|0123456789|0123456789|0123456789|0123456789|".getBytes());
+        // We can shift left amounts greater than the block size
+        buf.shiftBytesLeft(24, 24);
+        assertBuffer("01234569|0123456789|0123456789|".getBytes());
```

Comment:
Borrowing a convention from counting time in music, I think this makes the transpositions easier to see. 
```suggestion
        buf.writeBytes("0123456789|1123456789|2123456789|3123456789|4123456789|".getBytes());
        // We can shift left amounts greater than the block size
        buf.shiftBytesLeft(24, 24);
        assertBuffer("01234569|3123456789|4123456789|".getBytes());
```